src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/EventGridEventExtensions.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Azure.Messaging.EventGrid;
+using System;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions
+{
+    /// <summary>
+    /// This class offers extensions for EventGridEvents.
+    /// </summary>
+    public static class EventGridEventExtensions
+    {
+        private const string SyncTokenPropertyName = "syncToken";
+
+        /// <summary>
+        /// Tries to create the object from the details in object. Return value indicates whether the operation succeeded or failed.
+        /// </summary>
+        /// <param name="eventGridEvent"> EventGridEvent from EventGrid</param>
+        /// <param name="pushNotification"> If this method returns true, the push notification object contains details populated from eventGridEvent. If this method returns false, the pushNotification object is null. </param>
+        /// <returns></returns>
+        public static bool TryCreatePushNotification(this EventGridEvent eventGridEvent, out PushNotification pushNotification)
+        {
+            pushNotification = null;
+
+            if (eventGridEvent.Data == null || eventGridEvent.EventType == null || eventGridEvent.Subject == null)
+            {
+                return false;
+            }
+
+            if (!Uri.TryCreate(eventGridEvent.Subject, UriKind.Absolute, out Uri resourceUri))
+            {
+                return false;
+            }
+
+            try
+            {
+                JsonElement eventGridEventData = JsonDocument.Parse(eventGridEvent.Data.ToString()).RootElement;
+
+                if (eventGridEventData.ValueKind == JsonValueKind.Object &&
+                    eventGridEventData.TryGetProperty(SyncTokenPropertyName, out JsonElement syncTokenJson) &&
+                    syncTokenJson.ValueKind == JsonValueKind.String)
+                {
+                    pushNotification = new PushNotification()
+                    {
+                        SyncToken = syncTokenJson.GetString(),
+                        EventType = eventGridEvent.EventType,
+                        ResourceUri = resourceUri
+                    };
+
+                    return true;
+                }
+            }
```

Comment:
Since `JsonDocument.Parse` is the only call that could throw `JsonException`, I'd suggest wrapping only that line in the try block. Everything else can be outside the try block to make it easier to understand in future.